detectors/node/opentelemetry-resource-detector-docker/src/detectors/DockerDetector.ts

```diff
@@ -51,21 +53,39 @@ export class DockerCGroupV1Detector implements Detector {
 
   private async _getContainerId(): Promise<string | undefined> {
     try {
-      const rawData = await DockerCGroupV1Detector.readFileAsync(
+      let rawData = await DockerDetector.readFileAsync(
         this.DEFAULT_CGROUP_PATH,
         this.UTF8_UNICODE
       );
-      const splitData = rawData.trim().split('\n');
+      let splitData = rawData.trim().split('\n');
       for (const str of splitData) {
         if (str.length >= this.CONTAINER_ID_LENGTH) {
           return str.substring(str.length - this.CONTAINER_ID_LENGTH);
         }
       }
+
+      // If this code is reached - we then check for the V2_PATH
+      rawData = await DockerDetector.readFileAsync(
```

Comment:
Suggestion: this function is currently trying to get `containerId` for both V1 and V2.
I think it will improve the readability to split the function into 2: `_getContainerIdV1()` and `_getContainerIdV2()` and then call them from here:
```js
return _getContainerIdV1() ?? _getContainerIdV2();
```

The code is also very similar and you might have opportunities to reuse some of it for both cases

Reply:
I thought about it and it seemed natural to do that but overall it was almost the same code but I agree - it is more readable - will do.